apis/core/lib/ResourceStore.ts

```diff
@@ -110,31 +116,40 @@ export default class implements ResourceStore {
     this.__deletedGraphs = new TermSet()
   }
 
-  async get(id: string | Term | undefined): Promise<GraphPointer<NamedNode, DatasetExt> | undefined> {
-    if (!id) {
-      return undefined
-    }
-
-    let term: NamedNode
+  async get(id: string | Term | undefined, opts?: GetOptions): Promise<GraphPointer<NamedNode, DatasetExt>> {
+    let resource: GraphPointer<NamedNode, DatasetExt> | undefined
+    let term: NamedNode | undefined
     if (typeof id === 'string') {
       term = $rdf.namedNode(id)
-    } else if (id.termType === 'NamedNode') {
-      term = id
+    } else if (id && id.termType !== 'NamedNode') {
+      throw new Error('Loaded resource id must be NamedNode')
     } else {
-      return undefined
+      term = id
+    }
+
+    if (term) {
+      if (!this.__resources.has(term)) {
+        resource = await this.__storage.loadResource(term)
+        if (resource) {
+          this.__resources.set(term, resource)
+        }
+      }
+
+      resource = this.__resources.get(term)
     }
-    if (!this.__resources.has(term)) {
-      const resource = await this.__storage.loadResource(term)
-      if (resource) {
-        this.__resources.set(term, resource)
+
+    if (!resource) {
+      if (opts?.allowMissing) {
+        return undefined as any
       }
+      throw new NotFoundError(term)
```

Comment:
If I'm not mistaken, `NotFoundError` results in a 404 response from the API. I think in a lot of cases that's not what we want when loading a resource from the store. 404 should only happen when the resource at the requested URI doesn't exist, but not when another resource is not found while processing the request.

Reply:
Fair point 👌

Reply:
I see your fix, but now how about all the cases that *should* be 404? :D

Reply:
hydra-box will do that earlier, before it gets to our handlers

Reply:
Oh, alright then :)